test/GasAccounting.t.sol

```diff
@@ -99,307 +124,515 @@ contract MockGasAccounting is GasAccounting, Test {
-        // FIXME: fix before merging spearbit-reaudit branch
-        vm.skip(true);
-
         uint256 atlasBalance = 100 ether;
         uint256 borrow1 = 75 ether;
         uint256 borrow2 = 10 ether;
         uint256 borrow3 = 15 ether;
 
+        // Ensure the execution environment starts with zero balance
+        deal(executionEnvironment, 0);
+        mockGasAccounting.setLock(executionEnvironment, 0, uint8(ExecutionPhase.SolverOperation));
+
+        // Fund the contract with enough ETH (initial claims + total borrow amount)
+        uint256 totalBorrowAmount = borrow1 + borrow2 + borrow3;
         deal(address(mockGasAccounting), initialClaims + atlasBalance);
 
+        // Verify the contract has sufficient balance before borrowing
+        assertEq(
+            address(mockGasAccounting).balance, initialClaims + atlasBalance, "Contract should have enough ETH balance"
+        );
+
+        // Set the phase to an allowed phase for borrowing
+        mockGasAccounting.setPhase(ExecutionPhase.PreOps);
+
+        // Start borrowing operations
         vm.startPrank(executionEnvironment);
         mockGasAccounting.borrow(borrow1);
+        assertEq(
+            executionEnvironment.balance,
+            borrow1,
+            "Execution environment balance should equal borrow1 after first borrow"
+        );
+
         mockGasAccounting.borrow(borrow2);
+        assertEq(
+            executionEnvironment.balance,
+            borrow1 + borrow2,
+            "Execution environment balance should equal borrow1 + borrow2 after second borrow"
+        );
+
         mockGasAccounting.borrow(borrow3);
+        assertEq(
+            executionEnvironment.balance,
+            borrow1 + borrow2 + borrow3,
+            "Execution environment balance should equal borrow1 + borrow2 + borrow3 after third borrow"
+        );
         vm.stopPrank();
 
-        assertEq(executionEnvironment.balance, borrow1 + borrow2 + borrow3);
+        // Verify the final balance of the execution environment
+        assertEq(
+            executionEnvironment.balance,
+            borrow1 + borrow2 + borrow3,
+            "Final execution environment balance should equal total borrowed amount"
+        );
+
+        // Verify the final balance of the contract
+        uint256 expectedFinalContractBalance = initialClaims + atlasBalance - totalBorrowAmount;
+        assertEq(
+            address(mockGasAccounting).balance,
+            expectedFinalContractBalance,
+            "Final contract balance should be initial balance minus total borrowed amount"
+        );
+        tearDown();
     }
 
     function test_shortfall() public {
-        // FIXME: fix before merging spearbit-reaudit branch
-        vm.skip(true);
+        // Ensure the execution environment starts with zero balance
+        deal(executionEnvironment, 0);
+
+        // Set initial claims in the contract
+        mockGasAccounting.setClaims(initialClaims);
```

Comment:
Would be good to also try setting withdrawals, fees, writeoffs and deposits as well as claims, and checking the output is as expected given those inputs

Reply:
Added more tests